### ISSUE TITLE
Notifications icon does nothing without JavaScript

### ISSUE THREAD

**Issue by trichoplax** [codidact/qpixel#1833]
**Describe the bug**
With JavaScript disabled, clicking on either the envelope icon or the number of notifications in a blue circle next to the envelope icon has no effect (both are a link back to the current page). This prevents the user from viewing their notifications or marking them as read or unread.

**To Reproduce**
1. Go to any Codidact page while signed in.
2. Click on the envelope icon (top right, title text "Open notifications").
3. See nothing happens other than a `#` being added to the end of the current page URL.

**Expected behavior**
A user with JavaScript disabled should see a list of their notifications. This could be by various approaches. For example:
- [x] Make the envelope a link to the [notifications page](https://codegolf.codidact.com/users/me/notifications).
- [ ] Make the notifications panel a [popover](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API) so that it does not rely on JavaScript and users with JavaScript disabled can still use the panel in page rather than being directed to the separate notifications page.

Ideally there would also be a way of marking notifications as read or unread for users with JavaScript disabled. Currently the automatic marking of notifications as read and the manual toggling between read and unread both depend on JavaScript. Even if a user manages to navigate to the correct URL for the notifications page, they still cannot mark their messages as read without JavaScript. I don't know the best way to make these functionalities available. Potential approaches include:
- [ ] Make the links in the notifications page / notifications panel (depending on which is made to be accessible without JavaScript) include some indicator to the server that a notification should be marked as read. That is, following the link marks the notification as read, rather than having a separate request sent by JavaScript.
- [ ] Make the "mark read", "mark unread", and "Mark all as read" links in the notifications panel have fallback behaviour so they still work without JavaScript (or add them to the notifications page if the notifications panel remains unavailable without JavaScript). Without JavaScript this may require a page reload, whereas with JavaScript the change happens in page.

**Comment by Oaphi** [comment]
> Make the notifications panel a [popover](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API)

I don't think that's an option for now - most likely in a couple of years once browsers catch up - support is still a bit spotty (Firefox only since April 2024, Chrome and Safari since late 2023). This leaves us with some clever `<dialog>` workarounds or, barring that, with the easy way of just linking to the correct page.
